### PR TITLE
Add dockerfile, requirements and test for no ML build

### PR DESCRIPTION
**Pull request by antoniofilipovic**
### Description

The idea of this PR is to add a Dockerfile for build of MAGE without ML, which is often used when shipping customer images

### Pull request type

- [ ] Bugfix
- [ ] Algorithm/Module
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Related issues

Delete if this PR doesn't resolve any issues. Link the issue if it does.

######################################

#### Module/Algorithm
- [ ] Core algorithm/module implementation
- [ ] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Unit tests
- [ ] End-to-end tests
- [ ] Code documentation
- [ ] README short description
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)
- [ ] Update GQLALchemy signatures in [query builder](https://github.com/memgraph/gqlalchemy/blob/main/gqlalchemy/graph_algorithms/query_builder.py  ) using [query module signature generator](https://github.com/memgraph/gqlalchemy/blob/main/scripts/query_module_signature_generator.py)

######################################

### CODE REVIEW

**Review by antepusic — APPROVED** [review]
Having compared this with the existing Dockerfiles, I think this is fit for approving :+1:

File: Dockerfile.no_ML [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    python3-dev     `mage-memgraph` \
+    clang           `mage-memgraph` \
+    git             `mage-memgraph` \
+    unixodbc-dev        `mage-memgraph` \
+    --no-install-recommends \
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+COPY memgraph-${TARGETARCH}.deb .
+
+RUN dpkg -i memgraph-${TARGETARCH}.deb && rm memgraph-${TARGETARCH}.deb
```
Comment (MarkoBarisic) [inline Dockerfile.no_ML:33]: Dockerfile.release has a fix that was introduced in 1.12.1 release

`RUN dpkg -i memgraph-${TARGETARCH}.deb && rm memgraph-${TARGETARCH}.deb && rm /usr/lib/memgraph/query_modules/schema.so  \
&& rm /usr/lib/memgraph/query_modules/example_cpp.so`

Should this fix be applied here as well?

**Comment by MarkoBarisic** [comment]
Do you think tests for no-ML build should be added to `./github/workflows/test.yml` or the tests aren't needed because it can be implied from regular tests?

File: Dockerfile.no_ML (hunk 1 above)
Comment (antoniofilipovic) [reply Dockerfile.no_ML:33]: So, we don't need that here, because these modules cause an issue only for pytorch modules, and here we don't use them.

**Comment by antoniofilipovic** [comment]
> Do you think tests for no-ML build should be added to `./github/workflows/test.yml` or the tests aren't needed because it can be implied from regular tests?

Oh, nice idea. I think we should add tests most likely because that way we are sure no-ML image also works. Good catch

File: Dockerfile.no_ML (hunk 1 above)
Comment (MarkoBarisic) [reply Dockerfile.no_ML:33]: Thanks

**Review by MarkoBarisic — APPROVED** [review]
LGTM, let's merge EOD so I can make on demand platform builds available ASAP :muscle:

**Pull request merged**